packages/guideline-blocks-settings/src/hooks/useIsInViewport.ts

```diff
@@ -0,0 +1,36 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { RefObject, useEffect } from 'react';
+
+export const useIsInViewport = <T extends Element>({
+    ref,
+    disabled,
+    onChange,
+}: {
+    ref: RefObject<T>;
+    disabled: boolean;
+    onChange: (isInViewport: boolean) => void;
+}) => {
+    useEffect(() => {
+        if (disabled || !ref.current) {
+            return;
+        }
+
+        let isInViewport = false;
+
+        const intersectionObserver = new IntersectionObserver((entries) => {
+            for (const entry of entries) {
+                if (entry.isIntersecting !== isInViewport) {
+                    isInViewport = entry.isIntersecting;
+                    onChange(entry.isIntersecting);
+                }
```

Comment:
AFAIK this will ultimately mean that handleChange is called a lot more frequently than onChange is now, which doesn't make sense because the actual isIntersecting value is not changing. How it is now means that it will only call onChange when isIntersecting changes, but the callback provided to the intersectionObserver will be called anytime the intersection ratio changes, meaning that when you scroll way more callbacks will be triggered unneccessarily

Reply:
Ok perfect. I wasn't sure about it, and didn't dig deeper here :D